### PR TITLE
CGC 38 faq page

### PR DESCRIPTION
**Pull request by dstw-capco**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (avrx-capco) [inline .vscode/settings.json:2]: Is this file needed?

File: app/components/faq/AboutChallenge.component.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Grid, List, Header, Icon } from 'semantic-ui-react';
```
Comment (avrx-capco) [inline app/components/faq/AboutChallenge.component.js:2]: Icon is not used in this file, please remove the import.

File: app/components/faq/AboutChallenge.component.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import './style.scss';
+
+const AboutChallenge = () => (
+  <React.Fragment>
+    <Grid id="about-challenge" stackable centered className='question-section' style={{ paddingTop: '2rem' }}>
+      <Grid.Column width={12} className='bold-box' style={{ paddingBottom: '3rem', paddingTop: '2rem' }}>
```
Comment (avrx-capco) [inline app/components/faq/AboutChallenge.component.js:8]: Move inline styling into scss.

File: app/components/faq/AboutChallenge.component.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        <Header as='h1' textAlign='center'>What's the challenge about?</Header>
+        <List>
```
Comment (avrx-capco) [inline app/components/faq/AboutChallenge.component.js:10]: The content here isn't a list so it's [semantically wrong](https://www.w3schools.com/html/html5_semantic_elements.asp) to use <List/> and <List.item/>. Can you change this to <Container/> with <p/> or something similar?

File: app/components/faq/AboutChallenge.component.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Grid, List, Header, Icon } from 'semantic-ui-react';
+import './style.scss';
+
+const AboutChallenge = () => (
+  <React.Fragment>
```
Comment (avrx-capco) [inline app/components/faq/AboutChallenge.component.js:6]: This can be left empty `<>`

File: app/components/faq/EventRules.component.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Grid, List, Header } from 'semantic-ui-react';
+import './style.scss';
+
+const EventRules = () => (
+  <React.Fragment>
+    <Grid id="event-rules" stackable centered className='question-section' style={{ paddingTop: '2rem' }}>
+      <Grid.Column width={12} className='bold-box' style={{ paddingBottom: '3rem', paddingTop: '2rem' }}>
```
Comment (avrx-capco) [inline app/components/faq/EventRules.component.js:8]: Remove inline styling.

File: app/components/faq/EventRules.component.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Grid, List, Header } from 'semantic-ui-react';
+import './style.scss';
+
+const EventRules = () => (
+  <React.Fragment>
```
Comment (avrx-capco) [inline app/components/faq/EventRules.component.js:6]: `<>`

File: app/components/faq/FaqPage.component.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+            questionData={questionData}
+          />
+          <EventRules/>
+          <AboutChallenge/>
+          <SetupInstructions/>
+          <StravaSetupInstructions/>
+          <React.Fragment>
+            <Header as='h1' textAlign='center'>General</Header>
+            {questionData.general.map((question) => (
+                <QAndASection
```
Comment (avrx-capco) [inline app/components/faq/FaqPage.component.js:40]: Can you pass in the entire question object instead of 4 separate props?

File: app/components/faq/QAndAPair.component.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Grid } from 'semantic-ui-react';
+import './style.scss';
+
+const QAndAPair = (props) => (
```
Comment (avrx-capco) [inline app/components/faq/QAndAPair.component.js:5]: Instead of props you can just add `{ questionId, questionText, answerText }` so that you don't need `props.something`. Please add PropTypes.

File: app/components/faq/QuestionList.component.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { Grid, List, Header, Menu, Icon } from 'semantic-ui-react';
```
Comment (avrx-capco) [inline app/components/faq/QuestionList.component.js:2]: Remove unused imports: `Menu`, `Icon`

File: app/components/faq/QuestionList.component.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+import './style.scss';
+import { rightArrowIcon } from "./images";
+
+const QuestionList = (props) => (
```
Comment (avrx-capco) [inline app/components/faq/QuestionList.component.js:6]: Same as above, `{ questionData }` instead of props.

File: app/components/faq/QuestionList.component.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+  <React.Fragment>
+    <Grid stackable centered > 
+      <Grid.Column width={8} style={{ paddingBottom: '3rem' }}>
+        <List>
```
Comment (avrx-capco) [inline app/components/faq/QuestionList.component.js:10]: Do you need this `<List/>` wrapper? The content is already wrapped in a fragment.

File: app/components/faq/QuestionList.component.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+          <React.Fragment>
+            <Header><a href={"#event-rules"}>Event Rules</a></Header>
+            <Header><a href={"#about-challenge"}>What's the challenge about?</a></Header>
+            <Header><a href={"#set-up-instructions"}>Set Up instructions</a></Header>
+            <Header><a href={"#fitbit-specifc-steps"}>Fitbit Specific Steps</a></Header>
+            <Header ><a className="strava-hue" href={"#strava-specifc-steps"}>Strava Specific</a></Header>
+            <Header ><a className="strava-hue" href={"#strava-no-device"}>Strava Setup Guide (No Device/Manual entry)</a></Header>
+            <Header>General</Header>
+              {props.questionData.general.map((question) => (
+                <List.Item key={question.questionText}><img src={rightArrowIcon} alt="Next arrow" /> <a href={"#" + question.questionId}>{question.questionText}</a></List.Item>
```
Comment (avrx-capco) [inline app/components/faq/QuestionList.component.js:20]: Please use template literals when concatenating strings with variables.

File: app/components/faq/SetupInstructions.component.js [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+    <Header id="set-up-instructions" className="strava-hue" as='h1' textAlign='center'>Set up Instructions</Header>
+    <Grid className="step-text" stackable centered style={{ paddingTop: '2rem' }}>
+      <Grid.Column textAlign="left" verticalAlign='middle' width={6} style={{ paddingBottom: '3rem', paddingTop: '2rem' }}>
+        <List>
+          <List.Item>1. Connect your Fitbit or Strava account to Capco Challenge platform. Open the challenge website.</List.Item>
+          <List.Item>2. Enter your Capco 4-letter ID in the 'Register' section on this page for either Strava or Fitbit.</List.Item>
+        </List>
+      </Grid.Column>
+      <Grid.Column width={6}>
+        <img src={SetupInstructionsA}></img>
```
Comment (avrx-capco) [inline app/components/faq/SetupInstructions.component.js:21]: This can be simplified as ``. 
Can you add alt text? It's a required attribute.

File: app/components/faq/SetupInstructions.component.js [hunk 15]
```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { Grid, List, Header } from 'semantic-ui-react';
+import SetupInstructionsA from "./images/SetupInstructionsA.png";
```
Comment (avrx-capco) [inline app/components/faq/SetupInstructions.component.js:3]: Can you rename the images so that they start with lower case letter?

File: app/components/faq/SetupInstructions.component.js [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+    <Header id="fitbit-specifc-steps" as='h1' textAlign='center' className="fitbit-hue">Fitbit Specific Steps</Header>
+    <Grid className="step-text" stackable centered style={{ paddingBottom: '2rem', paddingTop: '2rem' }}>
+      <Grid.Column textAlign="left" verticalAlign='middle' width={6} style={{ paddingBottom: '3rem', paddingTop: '2rem' }}>
+        <List>
+          <List.Item>1. Link the Capco Global Step Challenge platform with your Fitbit account, by entering your Fitbit credentials.</List.Item>
+          <List.Item>2. If you haven’t set up a Fitbit account yet select “Sign Up” and follow these steps.</List.Item>
+        </List>
+      </Grid.Column>
+      <Grid.Column width={6}>
+        <img src={SetupInstructionsC}></img>
```
Comment (avrx-capco) [inline app/components/faq/SetupInstructions.component.js:44]: same as above

File: app/components/faq/images/index.js [hunk 17]
```diff
@@ -0,0 +1,3 @@
+import rightArrow from "./right-arrow.svg";
+
```
Comment (avrx-capco) [inline app/components/faq/images/index.js:2]: Can you add the other images as well?

File: app/components/faq/style.scss [hunk 18]
```diff
@@ -0,0 +1,74 @@
+@import '../../styles/variables.scss';
```
Comment (avrx-capco) [inline app/components/faq/style.scss:1]: Do you need this import?

File: app/components/faq/style.scss [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+  }
+}
+
+
+.question-list {
+    font-size: 16px;
+}
+
+.question-section {
+    padding-bottom: '2rem';
```
Comment (avrx-capco) [inline app/components/faq/style.scss:20]: You don't need quotes and these can be combined into `padding: 2rem 0 2rem 0;`

File: app/components/faq/FaqPage.component.js [hunk 20]
```diff
@@ -0,0 +1,75 @@
+/* eslint-disable react/jsx-no-target-blank */
+/* eslint-disable react/no-unescaped-entities */
+import React from 'react';
+import {
+    Segment, Header, Container, Icon
+} from 'semantic-ui-react';
+import QAndASection from './QAndAPair.component';
```
Comment (avrx-capco) [inline app/components/faq/FaqPage.component.js:7]: Can you import `QAndAPair` as `QAndAPair` instead of `QAndASection` or rename the component to `QAndASection`?

File: app/components/faq/QAndAPair.component.js (hunk 9 above)
Comment (dstw-capco) [reply app/components/faq/QAndAPair.component.js:5]: I passed the whole question in when using map in FaqPage.component.js, so instead using question.questionId now.

**Review by avrx-capco — APPROVED** [review]

**Comment by avrx-capco** [comment]
👏🏻✨

**Comment by dstw-capco** [comment]
Closing PR to wait for design from design team, then a new PR will be created once new design implemented.
